docs-samples/V4/JS/use-luis-for-language-understanding/luis-conversation-state.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Botbuilder v4 SDK - Using LUIS for language understanding
+ * 
+ * This bot demonstrates how to use LUIS as middleware and getting the intent from the turn context with 
+ * multiple turns using conversation state
+ * 
+ * To run this bot:
+ * 1) install these npm packages:
+ * npm install --save restify
+ * npm install --save botbuilder@preview
+ * npm install --save botbuilder-ai@preview
+ * 
+ * 2) From VSCode, open the package.json file and
+ * Update the property "main" to the name of the sample bot you want to run. 
+ *    For example: "main": "use-luis-for-language-understanding/luis-conversation-state.js" to run the this sample bot.
+ * 3) run the bot in debug mode. 
+ * 4) Load the emulator and point it to: http://localhost:3978/api/messages
+ * 5) Send the message "get weather" to engage with the bot.
+ *
+ */ 
+
+const { BotFrameworkAdapter, ConversationState, MemoryStorage, TurnContext } = require('botbuilder');
+const { LuisRecognizer } = require('botbuilder-ai');
+const { DialogSet } = require('botbuilder-dialogs');
+const restify = require('restify');
+
+// Create server
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`${server.name} listening to ${server.url}`);
+});
+
+// Create adapter
+const adapter = new BotFrameworkAdapter({ 
+    appId: process.env.MICROSOFT_APP_ID, 
+    appPassword: process.env.MICROSOFT_APP_PASSWORD 
+});
+
+const model = new LuisRecognizer({
+    // This appID is for a public app that's made available for demo purposes
+    // You can use it by providing your LUIS subscription key
+     appId: 'eb0bf5e0-b468-421b-9375-fdfb644c512e',
+    // replace subscriptionKey with your Authoring Key
+    // your key is at https://www.luis.ai under User settings > Authoring Key 
+    subscriptionKey: '<your subscription key>',
+    // The serviceEndpoint URL begins with "https://<region>.api.cognitive.microsoft.com", where region is the region associated with the key you are using. Some examples of regions are `westus`, `westcentralus`, `eastus2`, and `southeastasia`.
+    serviceEndpoint: 'https://westus.api.cognitive.microsoft.com'
+});
+
+
+adapter.use(model);
+
+// Add conversation state middleware
+const conversationState = new ConversationState(new MemoryStorage());
+adapter.use(conversationState);
+
+// Listen for incoming activities 
+server.post('/api/messages', (req, res) => {
+    // Route received activity to adapter for processing
+    adapter.processActivity(req, res, async(context) => {
+        if (context.activity.type === 'message') {
+            var utterance = context.activity.text;
+            
+            // Check topic flags in conversation state 
+            if (conversationState.weatherTopicStarted) 
```

Comment:
Should you be using the "convo" object like the other samples? Why are you accessing the "conversationState" object directly? I don't think "conversationstate.weatherTopicStarted" is saved to the bot's state nor is it written out to any database it is attached to. It's simply a global scope variable with the property "weatherTopicStarted" attached to it...but, since this is a flag, we should use the "topicStates" object instead?